_posts/2025-01-03-Clang-17-now-in-dev-container.markdown

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title:  "Clang 17 now in dev container, and other toolchain news"
+date:   2025-01-23
+categories: cheri toolchain
+author: Owen Anderson
+---
+
+Thanks to a collaboration with the [upstream CHERI toolchain](https://github.com/ctsrd-cheri/llvm-project), the CHERIoT toolchain has now been rebased onto Clang 17 from Clang 13, bringing us two years closer to upstream Clang.
+Major thanks to [Alex Richardson](https://github.com/arichardson) and [Sam Leffler](https://github.com/sleffler/) for their work on this effort!
+
+This updates brings with it substantially improved support for C++20 features, as well as preliminary support for some C++23 and C23 features.
+It also brings with it many improvements to the core of the RISC-V code generator, notably also benefiting code size: the firmware image for the Cheriot RTOS testsuite is 2.7% smaller when built with Clang 17 compared to Clang 13.
+Other highlights include compile time improvements, and too many under-the-hood fixes to enumerate.
+You can find detailed release notes for all Clang and LLVM releases on the LLVM.org [releases page](https://releases.llvm.org).
+
+## Other Toolchain Improvements
+
+Since landing the Clang 17 rebase, we've been busy bringing bugfixes and enhancements to the CHERIoT toolchain, including:
+
+### Language & Usability Improvements
+
+- [Implemented](https://github.com/CHERIoT-Platform/llvm-project/commit/025c5d452e8935ebbe2a09d78fb2a10c1c96a626) a new Clang diagnostic to warn on compartment exports that return void, or where the return value is unused.
```

Comment:
Maybe mention why this is useful?  Cross-compartment calls can fail in the switcher and there's no way of seeing that error if the result is void.

Reply:
Done.